oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -155,20 +155,20 @@ private ComputeEngineCredentials(ComputeEngineCredentials.Builder builder) {
   /** Clones the compute engine account with the specified scopes. */
   @Override
   public GoogleCredentials createScoped(Collection<String> newScopes) {
-    ComputeEngineCredentials.Builder builder =
-        this.toBuilder().setHttpTransportFactory(transportFactory).setScopes(newScopes);
-    return new ComputeEngineCredentials(builder);
+    return createScoped(newScopes, null);
   }
 
   /** Clones the compute engine account with the specified scopes and default scopes. */
   @Override
   public GoogleCredentials createScoped(
       Collection<String> newScopes, Collection<String> newDefaultScopes) {
     ComputeEngineCredentials.Builder builder =
-        ComputeEngineCredentials.newBuilder()
-            .setHttpTransportFactory(transportFactory)
-            .setScopes(newScopes)
-            .setDefaultScopes(newDefaultScopes);
+        (Builder)
+            this.toBuilder()
+                .setHttpTransportFactory(transportFactory)
+                .setScopes(newScopes)
+                .setDefaultScopes(newDefaultScopes)
+                .setAccessToken(null);
```

Comment:
I am trying to narrow the scope of this pr only fix this regression in behavior introduced by https://github.com/googleapis/google-auth-library-java/commit/7e268611d2c2152e84702b1c67ca846902bbe2d5. To matches the original behavior.
That said, in a follow up PR, I am considering moving the logic upstream to  `GoogleCredentials` or `OAuth2Credentials`. One thought is aside from createScoped(), does the same issue exists in creating copies and setting other properties differently, e.g. set a different universe. So perhaps  going to invalidate directly in `toBuilder()`. We can discuss further when I put up that PR.

> If a user passes in the same set of scopes (i.e. does not actually change the set of scopes for the credentials), this would invalidate the access token, right?

In this case, the old access token would probably work too, but it would be safer or at least no harm to invalidate it anyway and let the refresh workflow request the token from server again. 

> Do you think it would make sense to check to see if the scopes changed and only invalidate the access token then? If that makes sense, perhaps that could be done in GoogleCredentials so that it could apply to every single type of Credentials?

Hmm, let's discuss more in the follow-up pr, this will be a new check, and I'd rather introduce it in GoogleCredentials than individual credential classes if possible.

Reply:
Sounds good. We'll keep this scope for just fixing the bug and address everything else in a follow up PR.

> In this case, the old access token would probably work too, but it would be safer or at least no harm to invalidate it anyway and let the refresh workflow request the token from server again.

Yep makes sense. I just wondering what the expected behavior is from the user POV is. Also, wondering what the behavior from other language auth libraries. I think it makes sense to always invalidate the access token even if the access token is technically valid.